### DISCUSSION TITLE
Rust-based snappy node package doesn't work in Next.js deployed on Vercel

### DISCUSSION THREAD

**Discussion by KevinShiCA** [vercel/vercel#6940] [Help]
I have a Next.js app deployed on Vercel. I'd like to use [snappy](https://www.npmjs.com/package/snappy) in `getServerSideProps` for compression. Snappy uses @napi-rs to for the necessary bindings, i.e. the following is present in the package.json.

```
  "optionalDependencies": {
    "@napi-rs/snappy-win32-x64-msvc": "7.0.4",
    "@napi-rs/snappy-darwin-x64": "7.0.4",
    "@napi-rs/snappy-linux-x64-gnu": "7.0.4",
    "@napi-rs/snappy-linux-x64-musl": "7.0.4",
    "@napi-rs/snappy-linux-arm64-gnu": "7.0.4",
    "@napi-rs/snappy-win32-ia32-msvc": "7.0.4",
    "@napi-rs/snappy-linux-arm-gnueabihf": "7.0.4",
    "@napi-rs/snappy-darwin-arm64": "7.0.4",
    "@napi-rs/snappy-android-arm64": "7.0.4",
    "@napi-rs/snappy-freebsd-x64": "7.0.4",
    "@napi-rs/snappy-linux-arm64-musl": "7.0.4",
    "@napi-rs/snappy-win32-arm64-msvc": "7.0.4"
  }
```

Locally I have `@napi-rs/snappy-darwin-arm64` in node modules because I develop on a mac, but on my Vercel production deployment none of these were installed, resulting in `ERROR Error: Can not load bindings    at loadBinding (/var/task/node_modules/@node-rs/helper/lib/loader.js:50:11)`.

`@node-rs/helper` will log the list of installed packages if it failed to load bindings, and nothing was logged here leading me to believe that none of these optional dependencies are compatible with Vercel's production server.

Does anyone know offhand what kind of OS/arch Vercel runs for production Next.js deployments? I'm on Next.js 12 with SWC enabled if it matters.

**Comment by mikenikles** [comment]
I experience the same error with https://github.com/devtomio/sodium. @KevinShiCA any chance you've found a workaround since you posted the initial message?

**Comment by dreadera** [comment]
Experiencing the same with importing https://github.com/JaniAnttonen/winston-loki while importing it into my `/pages/api` code.
